pygsti/tools/matrixtools.py

```diff
@@ -31,37 +31,6 @@
 EXPM_DEFAULT_TOL = 2**-53  # Scipy default
 
 
-def trace(m):  # memory leak in numpy causes repeated trace calls to eat up all memory --TODO: Cython this
-    """
-    The trace of a matrix, sum_i m[i,i].
-
-    A memory leak in some version of numpy can cause repeated calls to numpy's
-    trace function to eat up all available system memory, and this function
-    does not have this problem.
-
-    Parameters
-    ----------
-    m : numpy array
-        the matrix (any object that can be double-indexed)
-
-    Returns
-    -------
-    element type of m
-        The trace of m.
-    """
-    return sum([m[i, i] for i in range(m.shape[0])])
-#    with warnings.catch_warnings():
-#        warnings.filterwarnings('error')
-#        try:
-#            ret =
-#        except Warning:
-#            print "BAD trace from:\n"
-#            for i in range(M.shape[0]):
-#                print M[i,i]
-#            raise ValueError("STOP")
-#    return ret
```

Comment:
This function is now inlined with ``np.trace``. The reported memory leak was fixed a long time ago.